contracts/src/v0.8/functions/dev/1_0_0/accessControl/TermsOfServiceAllowList.sol

```diff
@@ -102,23 +105,30 @@ contract TermsOfServiceAllowList is Routable, ITermsOfServiceAllowList {
     }
 
     // Add recipient to the allow list
-    s_allowedSenders[recipient] = true;
+    s_allowedSenders.add(recipient);
   }
 
   /**
    * @inheritdoc ITermsOfServiceAllowList
    */
-  function isAllowedSender(address sender) public view override returns (bool) {
+  function getAllAllowedSenders() external view override returns (address[] memory) {
```

Comment:
This allowlist contract is upgradeable, so I am not going to block the PR because of this, but we do need to add a start & end once our user count grows past a few thousand.